dask/array/creation.py

```diff
@@ -582,6 +582,90 @@ def tril(m, k=0):
     return Array(dsk, name, shape=m.shape, chunks=m.chunks, dtype=m.dtype)
 
 
+@wraps(np.pad)
+def pad(array, pad_width, mode, **kwargs):
+    array = da.asarray(array)
+
+    if mode not in ["constant", "edge", "linear_ramp"]:
+        raise NotImplementError("`pad` does not support the given `mode`.")
```

Comment:
There are various statistical paddings like `maximum`, `mean`, and `minimum`. These could be handled by computing on the global array (possibly sliced if `stat_length` is provided) and then treating them like the constant padding case.

The `median` case gets a little tricky. If we make some assumptions, it's probably possible to come up with a decent implementation here. That said, would help if those assumptions were informed by user feedback.

Probably `reflect` and `symmetric` work as is, but we would need to add a check to ensure the padding doesn't exceed what is within each chunk. We can probably error then and have the user fix this case or we can rechunk for them (possibly with a warning).

Getting something like `wrap` to work will require performing some global operations like slicing the Dask Array into pieces, moving them around, and running `block` on all of them to stitch it together into one Dask Array with the appropriate constraint. We could consider handling `reflect` and `symmetric` this way as well. Though the global information is not strictly needed in those cases.

Finally there is the possibility of a user supplying an arbitrary function. We probably could implement this and do have some thoughts here, but it is enough of a special case that we might want to wait and see if people ask for this feature and see what they expect from it.